optuna/pruners/hyperband.py

```diff
@@ -50,6 +50,9 @@ class HyperbandPruner(BasePruner):
         at least :math:`4 \\times 10` pruners are consumed for startup.
 
     Args:
+        max_resource:
+            A parameter for specifying the maximum resource allocated to a trial noted as :math:`R`
+            in the paper.
```

Comment:
I think the `max_resource` will be the same value of maximum iteration steps (e.g., `max_epoch` for neural network training) in the objective function. If you describe the relationship between them, it will be helpful for users.